test/src/test/java/jenkins/model/JenkinsInitializationTest.java

```diff
@@ -0,0 +1,98 @@
+package jenkins.model;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runners.model.Statement;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@Issue("JENKINS-50164")
+public class JenkinsInitializationTest {
+
+    private static final Map<String, String> PROPS_TO_RESTORE = new LinkedHashMap<>();
+    @Rule
+    public RestartableJenkinsRule story = new RestartableJenkinsRule();
+
+    @Rule
+    public LoggerRule loggerRule = new LoggerRule();
+
+    @BeforeClass
+    public static void beforeClass() {
+        Stream.of(Jenkins.BUILDS_DIR_PROP, Jenkins.WORKSPACES_DIR_PROP)
+                .forEach(property -> PROPS_TO_RESTORE.put(property, System.getProperty(property)));
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        PROPS_TO_RESTORE.forEach((k, v) -> {
+            if (v == null) {
+                System.clearProperty(k);
+            } else {
+                System.setProperty(k, v);
+            }
+        });
+    }
+
+    @Test
+    public void buildsDir() throws Exception {
+        loggerRule.record(Jenkins.class, Level.WARNING).capture(10);
+        story.addStep(new Statement() {
```

Comment:
Use `then`, which takes a SAM that can be implemented via lambda.

Reply:
done